VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -102,38 +116,71 @@ const QPoint ViewItem::positionOfNode(Model::Node *node) const
 	return QPoint(-1, -1);
 }
 
-QVector<QVector<Model::Node*>> ViewItem::nodesGetter()
+const QPoint ViewItem::positionOfItem(Item *item) const
 {
-	return nodes_;
+	auto vref = DCast<VNodeReference>(item);
+	if (vref)
+		return positionOfNode(vref->node());
+	else return QPoint(-1, -1);
 }
 
-void ViewItem::ensureColumnExists(int column)
+Model::Node* ViewItem::nodeAt(int column, int row)
 {
-	if (nodes_.size() <= column)
-		nodes_.resize(column + 1);
+	if (column < 0 || column >= nodes_.size())
+		return nullptr;
+	if (row < 0 || row >= nodes_[column].size())
+		return nullptr;
+	return nodes_[column][row];
 }
 
-QString ViewItem::debugNodes()
+void ViewItem::addEmptyNode(int column, int row, int desiredYPosOfNextItem)
```

Comment:
Like I mentioned before, there should be a more principled way to determining the size of the node, not just a direct integer.